doc/source/serve/key-concepts.md

```diff
@@ -24,91 +25,100 @@ class MyFirstDeployment:
   def __call__(self, request):
       return self.msg
 
-  def other_method(self, arg):
-      return self.msg
-
 my_first_deployment = MyFirstDeployment.bind("Hello world!")
+handle = serve.run(my_first_deployment)
+print(ray.get(handle.remote())) # "Hello world!"
 ```
 
-Deployments can be exposed in two ways: exposed to an end user over HTTP, or 
-exposed to other deployments in Python by including them in the input argument of `.bind()` for other deployments,
-e.g. `deployment_1.bind(deployment_2)`.
-By default, HTTP requests will be forwarded to the `__call__` method of the class (or the function) and a `Starlette Request` object will be the sole argument.
-You can also define a deployment that wraps a FastAPI app for more flexible handling of HTTP requests. See {ref}`serve-fastapi-http` for details.
+(serve-key-concepts-query-deployment)=
+
+## ServeHandle (composing deployments)
 
-To deploy multiple deployments that serve the same class or function, use the `name` option:
+Ray Serve enables flexible model composition and scaling by allowing multiple independent deployments to call into each other.
+When binding a deployment, included in the arguments can be references to _other bound deployments_.
+Then, at runtime this argument is converted to a {mod}`ServeHandle <ray.serve.handle.RayServeHandle>` that can be used to query the deployment using a Python-native API.
+Below is a basic example where the `Driver` deployment can call into two downstream models.
 
 ```python
-MyFirstDeployment.options(name="hello_service").bind("Hello!")
-MyFirstDeployment.options(name="hi_service").bind("Hi!")
+@serve.deployment
+class Driver:
+    def __init__(self, model_a_handle, model_b_handle):
+        self._model_a_handle = model_a_handle
+        self._model_b_handle = model_b_handle
+
+    async def __call__(self, request):
+        refa = await self._model_a_handle.remote(request)
+        refb = await self._model_b_handle.remote(request)
+        return (await refa) + (await refb)
+        
+
+model_a = ModelA.bind()
+model_b = ModelB.bind()
+
+# model_a and model_b will be passed to the Driver constructor as ServeHandles.
+driver = Driver.bind(model_a, model_b)
+
+# Deploys model_a, model_b, and driver.
+serve.run(driver)
```

Comment:
show the calling example? as well as moving these to doc_code?

Reply:
do you want to dive into the two types of handles? or that belong to a separate page?

Reply:
I think that should belong on a separate page(s). The dev workflow should talk about calling driver from python (cc @architkulkarni) and the model composition page should talk about calling deployments from one another (cc @shrekris-anyscale).